app/models/team_member.rb

```diff
@@ -0,0 +1,7 @@
+class TeamMember < ApplicationRecord
+  belongs_to :team
+  has_one :user
+
+  validates :role, presence: true
+  validates :team_id, presence: true
+end
```

Comment:
The 'has_one :user' association should be 'belongs_to :user' since a team member belongs to a user, not the other way around. This ensures the correct relationship and foreign key usage.